src/StaticOneTo.jl

```diff
@@ -0,0 +1,18 @@
+"""
+    StaticOneTo(N)
+
+A lazy and statically-sized analogue of `Base.OneTo(N)`.
+
+This iterator can only store the integers from 1 to `N`, so its
+`output_type_for_promotion` is `NoOutputType()`. An efficient method is provided
+for `unrolled_take`, but no other unrolled functions can use `StaticOneTo`s as
+output types.
+"""
+struct StaticOneTo{N} <: StaticSequence{N} end
```

Comment:
Is there a reason why we're not just using `StaticArrays.SOneTo`? (it's an exported/documented type)

Reply:
Ah, this subtypes StaticSequence

Reply:
`StaticOneTo` is used to define `StaticBitVector`. If we were to use `StaticArrays.SOneTo` instead, we would need to make `StaticArrays` a dependency of the package (right now it is just an extension). The definition of `StaticOneTo` is also just 5 lines, which seems like an acceptable tradeoff for avoiding a dependency.